Hush/Assets/Scripts/Game/SpellSelectorManager.cs

```diff
@@ -0,0 +1,65 @@
+using System;
+using UnityEngine;
+using Weapon;
+using Weapon.Enums;
+
+namespace Game
+{
+    public class SpellSelectorManager : MonoBehaviour
+    {
+        public static SpellSelectorManager Instance;
+        [SerializeField] private GameObject player;
+
+        private void Awake()
+        {
+            if (Instance == null)
+                Instance = this;
+            
+            DontDestroyOnLoad(Instance.gameObject);
+        }
+
+        public void SetupLightSpell(SpellType chosenSpell)
+        {
+            if (chosenSpell == SpellType.None)
+                chosenSpell = SpellType.InvisibleSpell; // default value
+            
+            GameManager.Instance.SetActiveLightSpell(chosenSpell);
+            ISpell spell = GetSpellComponent(chosenSpell);
+            GameManager.Instance.SetLightSpellCooldownTime(spell.LightCooldown);
+        }
+
+        public void SetupHeavySpell(SpellType chosenSpell)
+        {
+            if (chosenSpell == SpellType.None)
+                chosenSpell = SpellType.FireballSpell; // default value
+            
+            GameManager.Instance.SetActiveHeavySpell(chosenSpell);
+            ISpell spell = GetSpellComponent(chosenSpell);
+            GameManager.Instance.SetHeavySpellCooldownTime(spell.HeavyCooldown);
+        }
+
+        private ISpell GetSpellComponent(SpellType type)
+        {
+            ISpell spell;
+            switch (type)
+            {
+                case SpellType.FireballSpell:
+                    spell = player.GetComponent<FireballSpell>();
```

Comment:
You don't need this you can just GetComponent<ISpell> directly and it's going to return you whichever component matches

Reply:
Right now we have all of the Spells attached as Components to the player, so unless we detach and reattach them through this, wont there be confusion over which one it needs if I do something like GetComponent<ISpell> ? Not sure if this is what you mean?

Reply:
Makes sense! I guess that's okay then!